homeassistant/components/fivem/binary_sensor.py

```diff
@@ -0,0 +1,52 @@
+"""The FiveM binary sensor platform."""
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import FiveMEntity, FiveMEntityDescription
+from .const import DOMAIN, ICON_STATUS, NAME_STATUS
+
+
+class FiveMBinarySensorEntityDescription(
+    BinarySensorEntityDescription, FiveMEntityDescription
+):
+    """Describes FiveM binary sensor entity."""
+
+
+BINARY_SENSORS: tuple[FiveMBinarySensorEntityDescription, ...] = (
+    FiveMBinarySensorEntityDescription(
+        key=NAME_STATUS,
+        name=NAME_STATUS,
+        icon=ICON_STATUS,
```

Comment:
We normally don't need a custom icon when setting a device class. The device class will already have a custom icon in the frontend.